replicatedhq/ship
# Improve release naming, improve build idempotency

What I Did
------------

Ensure there is always a release name (Resolves #303)

How I Did it
------------

- Add a method on `api.ReleaseMetadata` `GetName()` that reads from `ChannelName` if present, otherwise `HelmMetadata.ChartName`, else `ship-`

- Also, some makefile tweaks so that `make build` can be the entrypoint, will only build the UI/tests if relevant files have changed.

How to verify it
------------

See #303 for repro steps, this chart now deploys successfully with `ship init`

Description for the Changelog
------------

Ensure `ship init` always provides a valid release name to a chart.

![](https://upload.wikimedia.org/wikipedia/commons/d/df/Ardent_class_silhouette.png)
